ENIGMAsystem/SHELL/Widget_Systems/None/nowidget_impl.cpp

```diff
@@ -69,25 +69,28 @@ class PasswordContext {
 };
 
 namespace enigma {
+
   bool widget_system_initialize() {
     return 0;
   }
 }
 
 namespace enigma_user {
 
-void show_error(string err, const bool fatal)
-{
-  printf("ERROR in some action of some event for object %d, instance id %d: %s\n",
-         (enigma::instance_event_iterator == NULL? enigma_user::global :
-            enigma::instance_event_iterator->inst == NULL? enigma_user::noone :
-              enigma::instance_event_iterator->inst->object_index),
-         (enigma::instance_event_iterator == NULL? enigma_user::global :
-            enigma::instance_event_iterator->inst == NULL? enigma_user::noone :
-              enigma::instance_event_iterator->inst->id),
-         err.c_str()
-    );
-  if (fatal) exit(0);
+void show_debug_message(string err, MESSAGE_TYPE type) {
+if (type == MESSAGE_TYPE::M_USER_ERROR || type == MESSAGE_TYPE::M_FATAL_USER_ERROR) {
+    printf((enigma::error_type(type) + ": in some action of some event for object %d, instance id %d: %s\n").c_str(),
+           (enigma::instance_event_iterator == NULL? enigma_user::global :
+              enigma::instance_event_iterator->inst == NULL? enigma_user::noone :
+                enigma::instance_event_iterator->inst->object_index),
+           (enigma::instance_event_iterator == NULL? enigma_user::global :
+              enigma::instance_event_iterator->inst == NULL? enigma_user::noone :
+                enigma::instance_event_iterator->inst->id),
+           err.c_str()
+      );
+  } else printf((enigma::error_type(type) + ": %s\n").c_str(), err.c_str()); 
```

Comment:
Is there a reason not to print event information for internal errors? It would probably help users track down the core problem.

Reply:
because it prints and object and instance the error spawned from and most these errors are from the engine not user code? Especially for info like opengl version and etc it seems confusing to print an instance.